beeai/agents/tools/text.py

```diff
@@ -111,6 +111,51 @@ async def _run(
         return StringToolOutput(result=f"Successfully inserted the specified text into {tool_input.file}")
 
 
+class InsertAfterSubstringToolInput(BaseModel):
+    file: AbsolutePath = Field(description="Absolute path to a file to edit")
+    insert_after_substring: str = Field(description="Substring to insert the text after")
+    new_string: str = Field(description="Text to insert")
+
+
+class InsertAfterSubstringTool(Tool[InsertAfterSubstringToolInput, ToolRunOptions, StringToolOutput]):
+    name = "insert_after_substring"
+    description = """
+    Inserts the provided text new_string on a new line after the first
+    occurrence of the specified substring insert_after_substring. The insertion
+    happens only once.
+    """
+    input_schema = InsertAfterSubstringToolInput
+
+    def _create_emitter(self) -> Emitter:
+        return Emitter.root().child(
+            namespace=["tool", "text", self.name],
+            creator=self,
+        )
+
+    async def _run(
+        self, tool_input: InsertAfterSubstringToolInput, options: ToolRunOptions | None, context: RunContext
+    ) -> StringToolOutput:
+        if not tool_input.insert_after_substring:
+            raise ToolError("No insertion was done because the specified substring wasn't provided")
```

Comment:
You could just type the argument as `NonEmptyString`:
https://github.com/packit/ai-workflows/blob/7cd29d3ce5209f38bdf57d72398f355cef86cae7/beeai/common/validators.py#L20